nsxt/resource_nsxt_edge_transport_node.go

```diff
@@ -671,43 +677,55 @@ func getTransportNodeFromSchema(d *schema.ResourceData, m interface{}) (*mpmodel
 	description := d.Get("description").(string)
 	displayName := d.Get("display_name").(string)
 	tags := getMPTagsFromSchema(d)
+	nodeID := d.Get("node_id").(string)
 	failureDomain := d.Get("failure_domain").(string)
 	hostSwitchSpec, err := getHostSwitchSpecFromSchema(d, m, nodeTypeEdge)
 	if err != nil {
 		return nil, fmt.Errorf("failed to create Transport Node: %v", err)
 	}
 
-	converter := bindings.NewTypeConverter()
-	var dataValue data.DataValue
-	var errs []error
+	var nodeDeploymentInfo *data.StructValue
+	if nodeID == "" {
+		/*
+			node_id attribute conflicts with node_deployment_info. As node_deployment_info is a complex object which has
+			attributes with default values, this schema property will always have values - therefore there is no simple way
+			to enforce this conflict within the provider (e.g check if node_id and node_deployment_info have values, then
+			fail.
+			So the provider will ignore node_deployment_info properties when node_id has a value - which would mean that
+			this edge appliance was created externally.
+		*/
```

Comment:
It might be helpful to also add a log statement here for debugging purposes.

e.g: "nodeID not specified, will deploy edge using values in deploymentConfig"

Reply:
Done